### PR TITLE
Ignore underscore in url

### LINKED ISSUE #14: Ignore paths (or templates) that start with _
Template finder treats every `*.html` and `*.md` as a template.

When layout files or partials are in the same directory they are being served as regular pages.

For example in Vanilla we have [`/templates/_layouts`](https://github.com/canonical-web-and-design/vanilla-framework/tree/master/templates/_layouts) and it results in unexpected files being served as pages:
- Docs layout: https://vanillaframework.io/_layouts/docs
- Site layout: https://vanillaframework.io/_layouts/site
- Empty root layout: https://vanillaframework.io/_layouts/_root
- Footer contents: https://vanillaframework.io/_layouts/_footer

Template finder should ignore paths with `_` (at the beginning of directory or file name).
If it's too strong as a default there should at least be an option to do so.

### PR DESCRIPTION
**Pull request by tbille**
# Done

Fixes #14 
If a url contains a _ at the start of one of its part then 404 since it's a include file

```
/_bla/bla      -> no
/_bla/_bla    -> no
/bla/_bla      -> no
/bla_/bla      -> yes
/bla/bla_      -> yes
/bla/bla        -> yes
```

# QA 

- Pull https://github.com/canonical-web-and-design/vanilla-framework
- replace the template finder module with this branch
- `git+git://github.com/tbille/canonicalwebteam.templatefinder@ignore-underscore#egg=canonicalwebteam.templatefinder`
- `./run`
- http://127.0.0.1:5000/_layouts/docs -> 404

### CODE REVIEW

**Comment by bartaz** [comment]
Had some trouble replacing the package locally. Ended up doing `pip uninstall canonicalwebteam.templatefiner` `git+git://github.com/tbille/canonicalwebteam.templatefinder@ignore-underscore#egg=canonicalwebteam.templatefinder`. But worked in the end.

File: setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     long_description_content_type="text/markdown",
     install_requires=[
         "Flask>=1.0",
-        "mistune>=0.8.4",
+        "mistune==0.8.4",
```
Comment (bartaz) [inline setup.py:19]: Is this change expected? It doesn't seem related to URL change.

**Review by bartaz — APPROVED** [review]
LGTM, one little comment about `mistune`.

File: setup.py (hunk 1 above)
Comment (tbille) [reply setup.py:19]: yeah this is the last version before their v2 which has big big breaking changes. So I pinned to the latest working version (which is the one that we currently use)

**Comment by tbille** [comment]
> Had some trouble replacing the package locally. Ended up doing pip uninstall canonicalwebteam.templatefiner git+git://github.com/tbille/canonicalwebteam.templatefinder@ignore-underscore#egg=canonicalwebteam.templatefinder. But worked in the end.

@bartaz sorry this is my fault I should have add better QA. I realize that by rereading them. Thanks for managing to run it. :)
